src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.Features.cs

```diff
@@ -236,6 +241,10 @@ internal void FastFeatureSet(Type key, object feature)
             {
                 throw new InvalidOperationException("Cannot set IISHttpContext in feature collection");
             };
+            if (key == IServerVariablesFeature)
+            {
+                throw new InvalidOperationException("Cannot set IServerVariablesFeature in feature collection");
```

Comment:
Most feature should be settable so they can be wrapped/shimmed. IISHttpContextType is the exception here as it's internal. Use a member field and initialize it to `this`.

Reply:
Updated this too